src/naemon/checks_service.c

```diff
@@ -486,6 +483,8 @@ int handle_async_service_check_result(service *temp_service, check_result *queue
 	/* was this check passive or active? */
 	temp_service->check_type = (queued_check_result->check_type == CHECK_TYPE_ACTIVE) ? CHECK_TYPE_ACTIVE : CHECK_TYPE_PASSIVE;
 
+	temp_service->latency = queued_check_result->latency;
```

Comment:
Why is this one needed? Should there be a similar place where the same is required for hosts?

Reply:
You are right, but for hosts this is done already here https://github.com/naemon/naemon-core/blob/master/src/naemon/checks_host.c#L407

Thing is, this does not make a huge difference for naemon-core alone because in naemon, the latency at the start of the check is the final latency already. But mod-gearman set the latency into the check-result when it's finished running a job and this latency can be higher.

Reply:
Right. Merlin probably does something similar, but can't remember. This should be fine for the core anyway.